src/Middlewares/ResponseTimeLogMiddleware.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Logcomex\PhpUtils\Middlewares;
+
+use Carbon\Carbon;
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Logcomex\PhpUtils\Dto\ResponseTimePayloadDto;
+use Logcomex\PhpUtils\Exceptions\BadImplementationException;
+
+/**
+ * Class ResponseTimeLogMiddleware
+ * @package Logcomex\PhpUtils\Middlewares
+ */
+class ResponseTimeLogMiddleware
+{
+    /**
+     * @var mixed
+     */
+    private $settings;
+
+    /**
+     * ResponseTimeLogMiddleware constructor.
+     */
+    public function __construct()
+    {
+        $this->settings = config('app');
+    }
+
+    /**
+     * @param Request $request
+     * @param Closure $next
+     * @return Response
+     * @throws BadImplementationException
+     */
+    public function handle(Request $request, Closure $next): Response
+    {
+        $response = $next($request);
+
+        if (defined('GLOBAL_FRAMEWORK_START')) {
+            $apiName = $this->getSetting('api-name');
+
+            if (empty($apiName)) {
+                throw new BadImplementationException(
+                    'PHU-007',
```

Comment:
Creio que o projeto tenha uma enumerador de erro.. ErroreEnum se não to enganado, se não tiver criado criar por favor